### PR TITLE
Remove ExpressionEncoder

### PR DESCRIPTION
**Pull request by karenfeng**
## What changes are proposed in this pull request?

Removes use of the non-public Spark API `ExpressionEncoder` in Glow to avoid breaking with API changes.

This PR also removes the private `vcfRowSchema` schema option as well as the accompanying converters.

## How is this patch tested?
- [x] Unit tests
- [ ] Integration tests
- [ ] Manual tests

### CODE REVIEW

File: core/src/test/scala/io/projectglow/vcf/VCFRowToVariantContextConverterSuite.scala [hunk 1]
```diff
@@ -30,14 +30,23 @@ import org.apache.commons.io.FileUtils
 import io.projectglow.common.{GenotypeFields, VCFRow}
 import io.projectglow.sql.GlowBaseTest
 
-class VCFRowToVariantContextConverterSuite extends GlowBaseTest with VCFConverterBaseTest {
+class VCFRowToVariantContextConverterSuite extends VCFConverterBaseTest {
```
Comment (henrydavidge) [inline core/src/test/scala/io/projectglow/vcf/VCFRowToVariantContextConverterSuite.scala:33]: Seems like we should rename this to `InternalRowToVariantContextConverter`

File: core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionGwas.scala [hunk 2]
```diff
@@ -43,6 +43,10 @@ case class LogitTestResults(
 
 object LogitTestResults {
   val nanRow: InternalRow = InternalRow(NaN, NaN, ArrayData.toArrayData(Seq(NaN, NaN)), NaN)
+  lazy val schema: StructType = ScalaReflection
```
Comment (henrydavidge) [inline core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionGwas.scala:46]: Is this only used in test code? If so please move to the test package.

Comment (henrydavidge) [reply core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionGwas.scala:46]: Btw, I don't think it's really any better to use `ScalaReflection` than `ExpressionEncoder`. It's an equally private interface.

Comment (karenfeng) [reply core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionGwas.scala:46]: This seems to be publicly documented: https://kb.databricks.com/data/schema-from-case-class.html

**Pull request merged**